web/src/components/navbar/NavBarPopupMenu.tsx

```diff
@@ -0,0 +1,187 @@
+import { ButtonIcon } from "@/components/ButtonIcon";
+import { NavMenuItem } from "@/components/navbar/NavMenuItem";
+import { Icon } from "@/components/njwds/Icon";
+import { AuthAlertContext } from "@/contexts/authAlertContext";
+import { AuthContext } from "@/contexts/authContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSelfRegister, onSignOut } from "@/lib/auth/signinHelper";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { getNavBarBusinessTitle, getUserNameOrEmail } from "@/lib/utils/helpers";
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { MenuItem, MenuList } from "@mui/material";
+import { useRouter } from "next/router";
+import { ReactElement, useContext } from "react";
+
+export type menuConfiguration =
+  | "profile"
+  | "profile-register-login"
+  | "profile-myNJ-logout"
+  | "login"
+  | "login-getstarted";
+
+interface Props {
+  handleClose: () => void;
+  open?: boolean;
+  hasCloseButton?: boolean;
+  menuConfiguration: menuConfiguration;
+}
+
+export const NavBarPopupMenu = ({
+  menuConfiguration,
+  handleClose,
+  open,
+  hasCloseButton,
+}: Props): ReactElement => {
+  const { userData, updateQueue } = useUserData();
+  const { dispatch } = useContext(AuthContext);
+  const { setRegistrationAlertStatus } = useContext(AuthAlertContext);
+
+  const router = useRouter();
+
+  const guestOrUserName =
+    menuConfiguration === "login" ? Config.navigationDefaults.navBarGuestText : getUserNameOrEmail(userData);
+  const isDashboardSelected = router.route === ROUTES.dashboard;
+  const isProfileSelected = router.route === ROUTES.profile;
+
+  const navBarBusinessTitle = getNavBarBusinessTitle(userData);
+
+  function handleListKeyDown(event: React.KeyboardEvent): void {
+    if (event.key === "Tab") {
+      event.preventDefault();
+      handleClose();
+    }
+  }
+
+  const loginMenuItem = (): ReactElement => {
+    return NavMenuItem({
```

Comment:
As disscussed briefly during stand up these are in this function call format rather than the normal component invocation because thats the main way you get focus for accessibility to work
![image](https://github.com/newjersey/navigator.business.nj.gov/assets/22485301/0abced62-7028-473d-b2d1-d86a71b08901)

https://github.com/mui/material-ui/issues/31006